File: bindings/prql-php/src/Message.php
<?php

declare(strict_types=1);

namespace Prql\Compiler;

/**
 * Compile result message.
 */
final class Message
{
    /**
     * Message kind. Currently only Error is implemented.
     */
    public MessageKind $kind;
    /**
     * Machine-readable identifier of the error.
     */
    public ?string $code;
    /**
     * Plain text of the error.
     */
    public string $reason;
    /**
     * A list of suggestions of how to fix the error.
     */
    public ?string $hint;
    /**
     * Character offset of error origin within a source file.
     */
    public ?Span $span;
    /**
     * Annotated code, containing cause and hints.
     */
    public ?string $display;
    /**
     * Line and column number of error origin within a source file.
     */
    public ?SourceLocation $location;
}

File: prqlc/bindings/dotnet/PrqlCompiler/Message.cs
namespace Prql.Compiler;

/// <summary>
/// Compile result message.
/// </summary>
public sealed record Message
{
    /// <summary>
    /// Message kind. Currently only Error is implemented.
    /// </summary>
    public MessageKind Kind { get; init; }

    /// <summary>
    /// Machine-readable identifier of the error. May be null.
    /// </summary>
    public string? Code { get; init; }

    /// <summary>
    /// Plain text of the error.
    /// </summary>
    public string Reason { get; init; } = string.Empty;

    /// <summary>
    /// A suggestion of how to fix the error. May be null.
    /// </summary>
    public string? Hint { get; init; }

    /// <summary>
    /// Character offset of error origin within a source file. May be null.
    /// </summary>
    public Span? Span { get; init; }

    /// <summary>
    /// Annotated code, containing cause and hints. May be null.
    /// </summary>
    public string? Display { get; init; }

    /// <summary>
    /// Line and column number of error origin within a source file. May be null.
    /// </summary>
    public SourceLocation? Location { get; init; }
}
